src/lib/utilities/Drawer/types.ts

```diff
@@ -0,0 +1,48 @@
+// Drawer Types
+
+export interface DrawerSettings {
+	open?: boolean;
+	/** A unique identifier, useful for setting contents. */
+	id?: string;
+	/** Set the anchor position.
+	 * @type {'left' | 'top' | 'right' | 'bottom'}
+	 */
+	/** Pass abitrary information for your own persona use. */
+	meta?: any;
+
+	// NOTE: the following must style in sync with the drawer props.
+
+	// --- Core ---
+	/** Set the anchor position.
+	 * @type {'left' | 'top' | 'right' | 'bottom'}
+	 */
+	position?: string;
```

Comment:
Since `position` is meant to only be one of these `'left' | 'top' | 'right' | 'bottom'` we should make the type a string union.
```suggestion
	/** Set the anchor position.
	 * @type {'left' | 'top' | 'right' | 'bottom'}
	 */
	position?: 'left' | 'top' | 'right' | 'bottom';
```